test/e2e/trafficcontrol_test.go

```diff
@@ -0,0 +1,310 @@
+		tc.Spec.TargetPort.Device = targetPort.(*v1alpha2.NetworkDevice)
+	case *v1alpha2.UDPTunnel:
+		if isTargetPortVXLAN {
+			tc.Spec.TargetPort.VXLAN = targetPort.(*v1alpha2.UDPTunnel)
+		} else {
+			tc.Spec.TargetPort.GENEVE = targetPort.(*v1alpha2.UDPTunnel)
+		}
+	case *v1alpha2.GRETunnel:
+		tc.Spec.TargetPort.GRE = targetPort.(*v1alpha2.GRETunnel)
+	case *v1alpha2.ERSPANTunnel:
+		tc.Spec.TargetPort.ERSPAN = targetPort.(*v1alpha2.ERSPANTunnel)
+	}
+
+	switch returnPort.(type) {
+	case *v1alpha2.OVSInternalPort:
+		tc.Spec.ReturnPort.OVSInternal = returnPort.(*v1alpha2.OVSInternalPort)
+	case *v1alpha2.NetworkDevice:
+		tc.Spec.ReturnPort.Device = returnPort.(*v1alpha2.NetworkDevice)
+	default:
+		tc.Spec.ReturnPort = nil
+	}
+
+	tc, err := data.crdClient.CrdV1alpha2().TrafficControls().Create(context.TODO(), tc, metav1.CreateOptions{})
+	require.NoError(t, err, "Failed to create TrafficControl")
+	return tc
+}
+
+func getOFPort(t *testing.T, data *TestData, portName string) int {
+	targetOFPort := -1
+	cmd := []string{"ovs-ofctl", "show", defaultBridgeName}
+	if err := wait.Poll(time.Second, 20*time.Second, func() (bool, error) {
+		stdout, _, err := data.RunCommandFromPod(antreaNamespace, antreaPodName, ovsContainerName, cmd)
+		if err != nil {
+			return false, nil
+		}
+		re := regexp.MustCompile(fmt.Sprintf(`(\d+)\(%s`, portName))
+		match := re.FindString(stdout)
+		if match != "" {
+			targetOFPort, _ = strconv.Atoi(strings.Split(match, "(")[0])
+			return true, nil
+		}
+		return false, nil
+	}); err != nil {
+		t.Logf("Not found expected OF port %s in OVS port list", portName)
+	}
+
+	return targetOFPort
+}
+
+func countMirroredPackets(t *testing.T, data *TestData, targetOFPort int) int {
+	var packets int
+	cmd := []string{"ovs-ofctl", "dump-flows", defaultBridgeName, "table=TrafficControl"}
+	stdout, _, _ := data.RunCommandFromPod(antreaNamespace, antreaPodName, ovsContainerName, cmd)
+	scanner := bufio.NewScanner(strings.NewReader(stdout))
+	scanner.Split(bufio.ScanLines)
+	re := regexp.MustCompile(fmt.Sprintf(`n_packets=(\d+).*load:0x%x\-\>NXM_NX_REG9`, targetOFPort))
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" {
+			continue
+		}
+		match := re.FindString(line)
+		if match == "" {
+			continue
+		}
+		match = strings.TrimLeft(match, "n_packets=")
+		match = strings.Split(match, ",")[0]
+		n, _ := strconv.Atoi(match)
+		packets += n
+	}
+	t.Logf("The total number of mirrored packets is %d", packets)
+	return packets
+}
+
+func countReceivedPackets(t *testing.T, data *TestData, targetPort string, isTargetPortTunnel bool) int {
+	var packets int
+	var stdout string
+	cmd := fmt.Sprintf("ifconfig %s", targetPort)
+	if isTargetPortTunnel {
+		stdout, _, _ = data.RunCommandFromPod(data.testNamespace, testCollectorPodName, agnhostContainerName, []string{"sh", "-c", cmd})
+	} else {
+		_, stdout, _, _ = data.RunCommandOnNode(testNode, cmd)
+	}
+
+	re := regexp.MustCompile(`RX packets[: ]\d+`)
```

Comment:
It can get submatch directly, please check examples.
```suggestion
	re := regexp.MustCompile(`RX packets[: ](\d+)`)
```

Reply:
Done.